Adding wit from OCI generates cmd error on dotnet build

When adding OCI wit, it generates an error when running `dotnet build` on the windows-plattform.

*Project.csproj*
```
<ItemGroup>
      <Wit Remove="**\*.wit" />
      <Wit Include="wit/wit.wasm" World="proxy" Registry="ghcr.io/webassembly/wasi/http:0.2.0" />
</ItemGroup>
```

*Output from dotnet build*
```console
Restore complete (0,4s)
  Project failed with 2 error(s) (1,3s)
    EXEC : error : Unable to write file
    C:\Users\USERNAME\.nuget\packages\bytecodealliance.componentize.dotnet.witbindgen\0.5.0-preview00008\build\BytecodeAlliance.Componentize.DotNet.WitBindgen.targets(60,9): error MSB3073: The command "C:\Users\USERNAME\.nuget\packages\bytecodealliance.componentize.dotnet.witbindgen\0.5.0-preview00008\build\..\tools\win-x64\wkg.exe oci pull ghcr.io/webassembly/wasi/http:0.2.0 --output wit;wit/wit.wasm" exited with code 1.

Build failed with 2 error(s) in 2,0s
```

It works fine if you run the the correct command once for the project.
```
C:\Users\USERNAME\.nuget\packages\bytecodealliance.componentize.dotnet.witbindgen\0.5.0-preview00008\build\..\tools\win-x64\wkg.exe oci pull ghcr.io/webassembly/wasi/http:0.2.0 --output wit/wit.wasm
```

So somewhere it adds `wit;` as a prefix to the bindget command.

--- a/.gitignore
+++ b/.gitignore
@@ -2,6 +2,6 @@ bin/
 obj/
 .vs/
 artifacts/
-test/WasmComponentSdkTest/testapps/OciWit/wit
+test/WasmComponentSdkTest/testapps/OciWit/wit/*.wasm
 *.binlog
 .DS_Store

--- a/Directory.Build.props
+++ b/Directory.Build.props
@@ -14,5 +14,8 @@
         <PrebuiltWitBindgenVersion>0.40.0</PrebuiltWitBindgenVersion>
         <PrebuildWkgVersion>0.10.0</PrebuildWkgVersion>
 
+        <!-- test artifacts -->
+        <WasmtimeVersion>30.0.2</WasmtimeVersion>
+
     </PropertyGroup>
 </Project>

--- a/src/WitBindgen/build/BytecodeAlliance.Componentize.DotNet.WitBindgen.targets
+++ b/src/WitBindgen/build/BytecodeAlliance.Componentize.DotNet.WitBindgen.targets
@@ -57,7 +57,7 @@
         </PropertyGroup>
         <Message Condition="'%(Wit.Registry)' != ''" Importance="high" Text="Pulling OCI %(Wit.Registry) to %(Wit.RelativeDir)..."  />
         <MakeDir Condition="'%(Wit.Registry)' != '' AND !Exists('%(Wit.RelativeDir)')" Directories="%(Wit.RelativeDir)" />
-        <Exec Condition="'%(Wit.Registry)' != ''" Command="$(WkgExe) oci pull %(Wit.Registry) --output $(WitOciFile)" /> 
+        <Exec Condition="'%(Wit.Registry)' != ''" Command="$(WkgExe) oci pull %(Wit.Registry) --output %(Wit.Identity)" /> 
     </Target>
 
     <Target Name="WitCompile_GetDependencies" DependsOnTargets="WitCompile_InvokeTool">

--- a/test/WasmComponentSdkTest/WasmComponentSdkTest/SimpleProducerConsumerTest.cs
+++ b/test/WasmComponentSdkTest/WasmComponentSdkTest/SimpleProducerConsumerTest.cs
@@ -41,7 +41,10 @@ public class SimpleProducerConsumerTest
     public void CanComposeImportWithExport()
     {
         var composed = FindModulePath("../testapps/SimpleConsumer", "composed.wasm");
-        var stdout = ExecuteCommandComponent(composed);
+        var (stdout, stderr, code) = ExecuteCommandComponent(composed);
+         if (code != 0) {
+            Assert.Fail(stderr);
+        }
         Assert.StartsWith("Hello, world on Wasm", stdout);
         Assert.Contains("123 + 456 = 579", stdout);
     }
@@ -50,15 +53,28 @@ public class SimpleProducerConsumerTest
     public void CanBuildAppFromOci()
     {
         var composed = FindModulePath($"../testapps/OciWit/bin/{Config}", "ociwit.wasm");
-        var stdout = ExecuteCommandComponent(composed);
+        var (stdout, stderr, code) = ExecuteCommandComponent(composed, "-S http");
+        if (code != 0) {
+            Assert.Fail(stderr);
+        }
         Assert.StartsWith("Oci is awesome!", stdout);
     }
 
-    private static string ExecuteCommandComponent(string componentFilePath)
+    private static (string, string, int) ExecuteCommandComponent(string componentFilePath)
     {
-        var startInfo = new ProcessStartInfo(WasmtimeExePath, $"-W component-model {componentFilePath}") { RedirectStandardOutput = true };
-        var stdout = Process.Start(startInfo)!.StandardOutput.ReadToEnd();
-        return stdout;
+        return ExecuteCommandComponent(componentFilePath, string.Empty);
+    }
+
+    private static (string, string, int) ExecuteCommandComponent(string componentFilePath, string wasmtime_args)
+    {
+        var startInfo = new ProcessStartInfo(WasmtimeExePath, $" {wasmtime_args} {componentFilePath}") { RedirectStandardOutput = true, RedirectStandardError = true };
+        var process = Process.Start(startInfo) ?? throw new InvalidOperationException("Failed to start process.");
+        process.WaitForExit();
+        int code = process.ExitCode;
+        var stdout = process.StandardOutput.ReadToEnd();
+        var stderr = process.StandardError.ReadToEnd();
+
+        return (stdout, stderr, code);
     }
 
     private static string GetWitInfo(string componentFilePath)
@@ -92,14 +108,17 @@ public class SimpleProducerConsumerTest
         if (matches.Count() == 1)
         {
             return Path.GetFullPath(matches.Single());
-        } 
-        else if (matches.Count() == 2 && matches.Any(x => Path.GetFullPath(x).Contains($"wasi-wasm\\native"))) {
-           return Path.GetFullPath(matches.First(x => Path.GetFullPath(x).Contains($"wasi-wasm\\native")));
         }
-        else if (matches.Count() == 2 && matches.Any(x => Path.GetFullPath(x).Contains($"wasi-wasm/native"))) {
-           return Path.GetFullPath(matches.First(x => Path.GetFullPath(x).Contains($"wasi-wasm/native")));
+        else if (matches.Count() == 2 && matches.Any(x => Path.GetFullPath(x).Contains($"wasi-wasm\\native")))
+        {
+            return Path.GetFullPath(matches.First(x => Path.GetFullPath(x).Contains($"wasi-wasm\\native")));
         }
-        else {
+        else if (matches.Count() == 2 && matches.Any(x => Path.GetFullPath(x).Contains($"wasi-wasm/native")))
+        {
+            return Path.GetFullPath(matches.First(x => Path.GetFullPath(x).Contains($"wasi-wasm/native")));
+        }
+        else
+        {
             throw new Exception($"Failed to get modules path, matched {matches.Count()} entries for directory {resolvedSearchDir} and filename {filename}.");
         }
     }

--- a/test/WasmComponentSdkTest/testapps/OciWit/Code.cs
+++ b/test/WasmComponentSdkTest/testapps/OciWit/Code.cs
@@ -1,18 +1,48 @@
 using CommandWorld.wit.exports.wasi.cli.v0_2_0;
 using MyFuncsWorld;
+using System.Text;
 
-public class RunImpl : IRun
+namespace ProxyWorld.wit.exports.wasi.http.v0_2_0
 {
-    public static void Run()
+    using ProxyWorld.wit.imports.wasi.http.v0_2_0;
+    public class IncomingHandlerImpl : IIncomingHandler
     {
-        Console.WriteLine("Oci is awesome!"); 
+        public static void Handle(ITypes.IncomingRequest request, ITypes.ResponseOutparam responseOut)
+        {
+            var content = Encoding.ASCII.GetBytes("Hello, from C#!");
+            var headers = new List<(string, byte[])> {
+            ("content-type", Encoding.ASCII.GetBytes("text/plain")),
+            ("content-length", Encoding.ASCII.GetBytes(content.Count().ToString()))
+        };
+            var response = new ITypes.OutgoingResponse(ITypes.Fields.FromList(headers));
+            var body = response.Body();
+            ITypes.ResponseOutparam.Set(responseOut, Result<ITypes.OutgoingResponse, ITypes.ErrorCode>.Ok(response));
+            using (var stream = body.Write())
+            {
+                stream.BlockingWriteAndFlush(content);
+            }
+            ITypes.OutgoingBody.Finish(body, null);
+        }
+    }
+}
+namespace CommandWorld.wit.exports.wasi.cli.v0_2_0
+{
+    public class RunImpl : IRun
+    {
+        public static void Run()
+        {
+            Console.WriteLine("Oci is awesome!");
+        }
     }
 }
 
-public class MyFuncsWorldImpl : IMyFuncsWorld
+namespace MyFuncsWorld
 {
-    public static int GetNumber()
+    public class MyFuncsWorldImpl : IMyFuncsWorld
     {
-        return 123;
+        public static int GetNumber()
+        {
+            return 123;
+        }
     }
 }

--- a/test/WasmComponentSdkTest/testapps/OciWit/Code.cs
+++ b/test/WasmComponentSdkTest/testapps/OciWit/Code.cs
@@ -1,9 +1,18 @@
 using CommandWorld.wit.exports.wasi.cli.v0_2_0;
+using MyFuncsWorld;
 
 public class RunImpl : IRun
 {
     public static void Run()
     {
         Console.WriteLine("Oci is awesome!"); 
+    }
+}
+
+public class MyFuncsWorldImpl : IMyFuncsWorld
+{
+    public static int GetNumber()
+    {
+        return 123;
     }
 }

--- a/test/WasmComponentSdkTest/testapps/OciWit/OciWit.csproj
+++ b/test/WasmComponentSdkTest/testapps/OciWit/OciWit.csproj
@@ -18,7 +18,8 @@
 
     <ItemGroup>
       <Wit Remove="**\*.wit" />
-      <Wit Include="wit/wit.wasm" World="command" Registry="ghcr.io/webassembly/wasi/cli:0.2.0" />
+      <Wit Include="wit"/>
+      <Wit Include="wit/cli.wasm" World="command" Registry="ghcr.io/webassembly/wasi/cli:0.2.0" />
     </ItemGroup>
 
 </Project>

--- a/test/WasmComponentSdkTest/testapps/OciWit/OciWit.csproj
+++ b/test/WasmComponentSdkTest/testapps/OciWit/OciWit.csproj
@@ -20,6 +20,7 @@
       <Wit Remove="**\*.wit" />
       <Wit Include="wit"/>
       <Wit Include="wit/cli.wasm" World="command" Registry="ghcr.io/webassembly/wasi/cli:0.2.0" />
+      <Wit Include="wit/http.wasm" World="proxy" Registry="ghcr.io/webassembly/wasi/http:0.2.0" />
     </ItemGroup>
 
 </Project>

--- a/test/WasmComponentSdkTest/testapps/OciWit/wit/api.wit
+++ b/test/WasmComponentSdkTest/testapps/OciWit/wit/api.wit
@@ -1,0 +1,5 @@
+package test:simple;
+
+world my-funcs {
+  export get-number: func() -> s32;
+}

--- a/test/WasmtimeCliFetcher/FetchWasmtime.targets
+++ b/test/WasmtimeCliFetcher/FetchWasmtime.targets
@@ -1,7 +1,5 @@
 <Project>
     <PropertyGroup>
-        <WasmtimeVersion>21.0.1</WasmtimeVersion>
-
         <WasmtimeTarget Condition="$([MSBuild]::IsOSPlatform('Windows'))">mingw</WasmtimeTarget>
         <WasmtimeTarget Condition="$([MSBuild]::IsOSPlatform('Linux'))">linux</WasmtimeTarget>
         <WasmtimeTarget Condition="$([MSBuild]::IsOSPlatform('OSX'))">macos</WasmtimeTarget>

